### PR TITLE
Fix JUnit retry implementation

### PR DESCRIPTION
**Pull request by sbabcoc**

### CODE REVIEW

File: example-junit/src/test/resources/META-INF/services/com.nordstrom.automation.junit.JUnitRetryAnalyzer [hunk 1]
```diff
@@ -0,0 +1 @@
+com.epam.reportportal.example.junit.extension.JUnitRetryImpl
```
Comment (sbabcoc) [inline example-junit/src/test/resources/META-INF/services/com.nordstrom.automation.junit.JUnitRetryAnalyzer:1]: This is the declaration of the retry analyzer implementation

File: example-junit/src/test/java/com/epam/reportportal/example/junit/extension/JUnitRetryImpl.java [hunk 2]
```diff
@@ -3,10 +3,10 @@
 import com.nordstrom.automation.junit.JUnitRetryAnalyzer;
 import org.junit.runners.model.FrameworkMethod;
 
-public class RetryImpl implements JUnitRetryAnalyzer {
+public class JUnitRetryImpl implements JUnitRetryAnalyzer {
 
 	@Override
 	public boolean retry(FrameworkMethod method, Throwable thrown) {
-		return thrown != null;
+		return true;
```
Comment (sbabcoc) [inline example-junit/src/test/java/com/epam/reportportal/example/junit/extension/JUnitRetryImpl.java:10]: With this hardcoded result, failed tests will always be retried. In this instance, we could also limit the retriable tests to the `RetryTest` class:
```java
@Override
public boolean retry(FrameworkMethod method, Throwable thrown) {
    return method.getDeclaringClass() == RetryTest.class;
}
```

File: example-junit/src/test/java/com/epam/reportportal/example/junit/extension/RetryTest.java [hunk 3]
```diff
@@ -1,22 +1,39 @@
...
 	private static final Logger LOGGER = LoggerFactory.getLogger(RetryTest.class);
 
 	private static final int RETRY_NUMBER = 20;
 	private static final AtomicInteger COUNTER = new AtomicInteger(0);
-
-//	@Ignore
+	
+	@BeforeClass
+	public static void beforeClass() {
+		System.setProperty(JUnitSettings.MAX_RETRY.key(), "3");
```
Comment (sbabcoc) [inline example-junit/src/test/java/com/epam/reportportal/example/junit/extension/RetryTest.java:27]: The automatic retry feature of `JUnit Foundation` is activated by setting the `junit.max.retry` System property to a positive integer value. This example isn't the typical way to specify this, however. Generally, you'd specify the maximum retry attempts in the `junit.properties` configuration file or on the command line.

File: example-junit/pom.xml [hunk 4]
```diff
@@ -74,12 +103,6 @@
                         -javaagent:"${settings.localRepository}/org/aspectj/aspectjweaver/${aspectj.version}/aspectjweaver-${aspectj.version}.jar"
                         -javaagent:${com.nordstrom.tools:junit-foundation:jar}
                     </argLine>
-                    <properties>
-                        <property>
-                            <name>listener</name>
-                            <value>com.nordstrom.automation.junit.RunAnnouncer</value>
-                        </property>
-                    </properties>
```
Comment (sbabcoc) [inline example-junit/pom.xml:82]: This listener specification was the source of the NullPointerException failures reported in [JUnit Foundation issue #62](https://github.com/Nordstrom/JUnit-Foundation/issues/62).

**Review by HardNorth — APPROVED** [review]
Well, thank you very much for the clarification!

**Pull request merged**